ftw/linkchecker/command/checking_links.py

```diff
@@ -1,35 +1,86 @@
-from zope.component.hooks import setSite
-from Testing.makerequest import makerequest
 from AccessControl.SecurityManagement import newSecurityManager
-from zope.globalrequest import setRequest
-import AccessControl
+from Products.CMFPlone.interfaces import IPloneSiteRoot
 from zc.relation.interfaces import ICatalog
 from zope.component import getUtility
+from zope.component.hooks import setSite
+import AccessControl
 
 
-def main(plone, *args):
+def get_plone_sites_information(app):
+    plone_sites = [obj for obj in app.objectValues(
```

Comment:
The plone sites are usually nested in to mount-points (directories). so this script should follow directories too. --> ask @Nachtalb

Reply:
@busykoala here is an example of how I do it, if you have any questions, feel free to ask.

https://github.com/4teamwork/bgbern.ng/blob/dbc4612e77617727d27e2c275b9187b65804c5ca/bgbern/ng/utils.py#L65-L84

Reply:
I adapted my function to the given example.
Should we, like it's done in the given example, later only get PloneSites without self? For developing reasons I'll leave that out now.

Please move the discussion to #16 so we can move on.